visualization/app/codeCharta/util/deltaGenerator.ts

```diff
@@ -12,15 +12,27 @@ export class DeltaGenerator {
 			if (path === CodeChartaService.ROOT_PATH) {
 				rootNode = node
 			} else {
-				const parentNode = getParent(hashMapWithAllNodes, path)
-				parentNode.children.push(node)
+				const childNode = node
+				let parentNode = getParent(hashMapWithAllNodes, path)
+				const originalParent = parentNode
+				const added = childNode.deltas["addedFiles"] ?? 0
+				const removed = childNode.deltas["removedFiles"] ?? 0
+
+				while (parentNode) {
```

Comment:
This will likely work, but it's not ideal performance wise. It should be possible to count together all files in a linear fashion when doing this in a depth first for loop.

Reply:
Any explicit suggestions are welcomed. I cannot find a better solution at this time.